src/ClaimService.Validation/Claim/EditClaimRequestValidator.cs

```diff
@@ -0,0 +1,168 @@
+      },
+      CascadeMode.Stop);
+
+    AddFailureForPropertyIf(
+      nameof(EditClaimRequest.Content),
+      x => x == OperationType.Replace,
+      new()
+      {
+        { x => !string.IsNullOrEmpty(x.value?.ToString().Trim()), EditClaimRequestValidatorResourses.NotEmptyContent },
+        { x => x.value.ToString().Trim().Length < 2000, EditClaimRequestValidatorResourses.TooLongContent }
+      },
+      CascadeMode.Stop);
+
+    await AddFailureForPropertyIfAsync(
+      nameof(EditClaimRequest.CategoryId),
+      x => x == OperationType.Replace,
+      new()
+      {
+        {
+          async (x) =>
+          {
+            return Guid.TryParse(x.value.ToString(), out Guid categoryId)
+              ? await _categoryRepository.DoesExistAsync(categoryId) &&
+              (dbClaim.Status == ClaimStatus.New ||
+              dbClaim.Status == ClaimStatus.Created ||
+              dbClaim.Status == ClaimStatus.Denied)
+              : false;
+          },
+          EditClaimRequestValidatorResourses.NotExistingCategoryId
+        }
+      },
+      CascadeMode.Stop);
+
+    AddFailureForPropertyIf(
+      nameof(EditClaimRequest.Status),
+      x => x == OperationType.Replace,
+      new()
+      {
+        {
+          x => Enum.TryParse(x.value?.ToString(), out ClaimStatus res) &&
+          Enum.IsDefined(typeof(ClaimStatus),res),
+          EditClaimRequestValidatorResourses.IncorrectClaimStatus
+        },
+        { x => Enum.TryParse(x.value?.ToString(), out ClaimStatus res) &&
+          res == ClaimStatus.Closed || res == ClaimStatus.Returned ? dbClaim.CreatedBy == senderId : true,
+          EditClaimRequestValidatorResourses.IncorrectUser
+        }
+      },
+      CascadeMode.Stop);
+
+    AddFailureForPropertyIf(
+      nameof(EditClaimRequest.Priority),
+      x => x == OperationType.Replace,
+      new()
+      {
+        {
+          x => Enum.TryParse(x.value?.ToString(), out ClaimPriority res) &&
+          Enum.IsDefined(typeof(ClaimPriority),res),
+          EditClaimRequestValidatorResourses.IncorrectClaimPriority
+        }
+      },
+      CascadeMode.Stop);
+
+    AddFailureForPropertyIf(
+      nameof(EditClaimRequest.Deadline),
+      x => x == OperationType.Replace,
+      new()
+      {
+        { x => string.IsNullOrEmpty(x.value?.ToString())
+          ? true
+          : DateTime.TryParse(x.value.ToString(), out DateTime result) &&
+          result > DateTime.Now,
+          EditClaimRequestValidatorResourses.IncorectDeadLineFormat
+        }
+      },
+      CascadeMode.Stop);
+
+  }
+  public EditClaimRequestValidator(
+    ICategoryRepository categoryRepository,
+    IClaimRepository claimRepository,
+    IHttpContextAccessor contextAccessor)
+  {
+    _categoryRepository = categoryRepository;
+    _claimRepository = claimRepository;
+    _contextAccessor = contextAccessor;
+
+    Guid senderId = _contextAccessor.HttpContext.GetUserId();
```

Comment:
Why do you use _contextAccessor instead of contextAccessor? May be remove excess _claimRepository  and _contextAccessor variables.